packages/notifi-react/lib/context/NotifiTargetContext.tsx

```diff
@@ -661,6 +668,151 @@ export const NotifiTargetContextProvider: FC<PropsWithChildren> = ({
     [],
   );
 
+  const getSignature = useCallback(
+    async (message: Uint8Array | string) => {
+      let signature: Uint8Array | string = '';
+
+      // TODO: Add logic for rest of the chains
+      switch (walletWithSignParams.walletBlockchain) {
+        case 'AVALANCHE':
+        case 'ETHEREUM':
+          signature = await walletWithSignParams.signMessage(
+            message as Uint8Array,
+          );
+          break;
+        default:
+          signature = '';
+      }
+      return reformatSignatureForWalletTarget(signature);
+    },
+    [walletWithSignParams.signMessage],
+  );
+
+  // NOTE: Temporarily commenting out this function because it will be needed later
+  // const xip43Impl = async () => {
+
+  //   const targetId = targetData?.wallet?.data?.id ?? '';
+  //   const address = '';
+
+  //   // TODO: get senderAddress from target
+  //   const senderAddress = '0xE80E42B5308d5b137FC137302d571B56907c3003';
+  //   const timestamp = Date.now();
+  //   const message = createConsentMessage(senderAddress, timestamp);
+  //   const signature = '';
+
+  //   if (!signature) {
+  //     throw Error('Unable to sign the wallet. Please try again.');
+  //   }
+
+  //   await frontendClient.verifyXmtpTarget({
+  //     input: {
+  //       web3TargetId: targetId,
+  //       accountId: address,
+  //       consentProofSignature: signature as string,
+  //       timestamp: timestamp,
+  //       isCBW: true,
+  //     },
+  //   });
+  //   // await signCoinbaseSignature(address, senderAddress);
+  //   await frontendClient.verifyCbwTarget({
+  //     input: {
+  //       targetId: targetId,
+  //     },
+  //   });
+  // };
+
+  const xmtpXip42Impl = useCallback(async () => {
+    const options: any = {
+      persistConversations: false,
+      env: 'production',
+    };
+    const address = walletWithSignParams.walletPublicKey;
+
+    const signer = {
+      getAddress: (): Promise<string> => {
+        return new Promise((resolve) => {
+          resolve(address);
+        });
+      },
+      signMessage: async (message: Uint8Array | string): Promise<string> => {
+        return getSignature(message);
+      },
+    };
+
+    const client = await xmtp.initialize({ options, signer });
+
+    if (client === undefined) {
+      throw Error('XMTP client is uninitialized. Please try again.');
+    }
+
+    // TODO: get senderAddress from target
+    const senderAddress = '0xE80E42B5308d5b137FC137302d571B56907c3003';
```

Comment:
Just wondering what is this address? does this one represent `notifi`?? or Tenant specific

Reply:
@kai-notifi, I think it's more of your question.

Reply:
This is tenant specific just for gmx. In the near future we should take this from the web3 target. I'm currently working on that so we can support that

Reply:
After disucssing with Kai, the sender address will be eventually retrieve from `Web3Target` ( `targetData.wallet.senderAddress` ). 
Just for VIZ CC: @ankush-notifi @kai-notifi

Reply:
It means this feature will not be able to use for other clients except GMX. so we might want to add a comment to make it clear 🙏
Others LGTM . Thanks for the great effort @ankush-notifi

Reply:
Feel free to merge after adding the comments for this section. And please also check it can be built and also functionally working on GMX. Thanks again! ✦ @ankush-notifi